### PR TITLE
Air puffs

### PR DESCRIPTION
**Pull request by austinhoag**
This PR is to create 3 new databases for the U19 Air Puffs task. The `puffs_acquisition` db is for keeping track of which hdf5 files have been ingested already, `puffs_lab` schema contains the cohort information, and the `puffs_behavior` db holds the session, trial, trial phase and puff information. I did not make any changes to the existing schemas (the newline changes are accidental and we can just use the master branch version of those files). The daily ingestion script is not included in this PR. Should it be?

### CODE REVIEW

**Comment by Alvalunasan** [comment]
-I think the ingestion script should be included in the PR as well. (At least these are included on the U19-pipeline-Matlab repo).
-I think puffs_acquisition & puffs_lab schemas should be part of a single schema (or even all 3 schemas part of a single one).
Does your ingestion process work like this?

1. You insert cohort & file_acquisition records from the spreadsheet you have mentioned in previous meetings.
2.  You run your ingestion code, fetch all **PuffsFileAcquisition** records, if the acquisition has not being ingested you insert every table:
**session_started, session, puff-session, puff-trial,  trial-phase & puff.**
3. If all insertions ran without issue you update the ingested flag for the current **PuffsFileAcquisition**.

**Comment by austinhoag** [comment]
OK, I added the current ingestion script to this PR. The script works just like how you expected. Yes, I can put the new tables all in one schema.

**Comment by Alvalunasan** [comment]
I think it looks great !!
My final comment would be:
Do you think **PuffCohort** insertion should be added as well in the h5 files loop?
Sorry, maybe you  explained this cohort insertion before but since you are inserting everything in the ingestion script I think is no harm to add that one as well in here.

**Comment by austinhoag** [comment]
Thanks for catching that. I added the line to insert into the `PuffsCohort()` table.

File: u19_pipeline/puffs.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    rig                         : tinyint                      # an integer that describes which rig was used. 0 corresponds to location = "pni-ltl016-05", 1 corresponds to location = "wang-behavior"
+    notes = ''                  : varchar(1024)                # notes recorded by experimenter during the session
+    stdout = NULL               : blob                         # stdout for the GUI during the session
+    stderr = NULL               : blob                         # stderr for the GUI during the session
+    sync = NULL                 : blob                         # At the start of the session, the software runs multiple python processes. This column contains the times of these processes in seconds 
+    """
+
+    class Trial(dj.Part):
+    	definition = """
+        -> PuffsSession
```
Comment (shenshan) [inline u19_pipeline/puffs.py:45]: -> master

File: u19_pipeline/puffs.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        trial_idx            : int                          # trial index, keep the original number in the file
+        ---
+		-> task.TaskLevelParameterSet.proj(level="level")   # The difficulty level of the trial
```
Comment (shenshan) [inline u19_pipeline/puffs.py:48]: Why do we need the `proj()` here?

File: u19_pipeline/puffs.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        ---
+	    phase_rel_start      : float                        # start time of the phase relative to the beginning of the session [seconds]
+	    phase_rel_finish     : float                        # end time of the phase relative to the beginning of the session [seconds]
+        """
+    
+    class Puff(dj.Part):
+    	definition = """
+        -> master.Trial
+        puff_idx             : tinyint                     # the index of the puff in this particular trial
+        side                 : tinyint                     # 0 = left side, 1 = right side
```
Comment (shenshan) [inline u19_pipeline/puffs.py:78]: does side have to be in the primary key? `puff_idx` is independently indexed on each side?

File: u19_pipeline/puffs.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    project_name         : varchar(64)                  # Corresponds to the path on bucket /puffs/netid/project_name/cohortX/ ...
+    cohort               : varchar(64)                  # Corresponds to the path on bucket /puffs/netid/project_name/cohortX/ ...
+    ---
+    """
+
+@schema
+class PuffsFileAcquisition(dj.Manual):
+    definition = """
+    -> PuffsCohort
+    rig                  : tinyint                      # an integer that describes which rig was used. 0 corresponds to location = "pni-ltl016-05", 1 corresponds to location = "wang-behavior"
```
[the remark this answers is not in the document]
Comment (austinhoag) [reply u19_pipeline/puffs.py:21]: Good idea, I will create a lookup table.

File: u19_pipeline/puffs.py (hunk 2 above)
Comment (austinhoag) [reply u19_pipeline/puffs.py:48]: good point we don't. It was left over from when I was renaming level, but I changed it back.

File: u19_pipeline/puffs.py (hunk 3 above)
Comment (austinhoag) [reply u19_pipeline/puffs.py:78]: Good point. Side does not have to be a primary key.

File: u19_pipeline/puffs.py [hunk 5]
```diff
@@ -18,7 +32,7 @@ class PuffsCohort(dj.Manual):
 class PuffsFileAcquisition(dj.Manual):
     definition = """
     -> PuffsCohort
-    rig                  : tinyint                      # an integer that describes which rig was used. 0 corresponds to location = "pni-ltl016-05", 1 corresponds to location = "wang-behavior"
+    -> Rig.rig
```
Comment (shenshan) [inline u19_pipeline/puffs.py:35]: rig.Rig?

File: u19_pipeline/puffs.py [hunk 6]
```diff
@@ -33,7 +47,7 @@ class PuffsSession(dj.Manual):
     -> acquisition.Session
     ---
     session_params = NULL       : blob                         # The parameters for this session, e.g. phase_durations, whether puffs are on, etc... 
-    rig                         : tinyint                      # an integer that describes which rig was used. 0 corresponds to location = "pni-ltl016-05", 1 corresponds to location = "wang-behavior"
+    -> Rig.rig
```
Comment (shenshan) [inline u19_pipeline/puffs.py:50]: rig.Rig

File: u19_pipeline/puffs.py (hunk 5 above)
Comment (austinhoag) [reply u19_pipeline/puffs.py:35]: Thanks for catching that.

File: u19_pipeline/puffs.py (hunk 6 above)
Comment (austinhoag) [reply u19_pipeline/puffs.py]: fixed

**Review by Alvalunasan — APPROVED** [review]
Awesome work Austin !!

**Pull request merged**